pkg/network/usm/sharedlibraries/watcher_test.go

```diff
@@ -303,76 +222,38 @@ func (s *SharedLibrarySuite) TestSoWatcherLeaks() {
-			return false
-		}
-
-		// Checking the PID associated with the 2 paths.
-		return checkPIDAssociatedWithPathID(watcher, fooPathID1, uint32(command1.Process.Pid)) &&
-			checkPIDAssociatedWithPathID(watcher, fooPathID2, uint32(command1.Process.Pid))
+		// Checking register callback was executed once for each library
+		// and that we're tracking the two command PIDs
+		return registerRecorder.CallsForPathID(fooPathID1) == 1 &&
+			registerRecorder.CallsForPathID(fooPathID2) == 1 &&
+			hasPID(watcher, command1)
 	}, time.Second*10, time.Second, "")
 
-	command2 := exec.Command(clientBin, fooPath1)
-	require.NoError(t, command2.Start())
-	registerProcessTerminationUponCleanup(t, command2)
+	command2 := fileopener.OpenFromAnotherProcess(t, fooPath1)
 
 	require.Eventuallyf(t, func() bool {
-		// Checking both paths exist.
-		if checkPathIDDoesNotExist(watcher, fooPathID1) || checkPathIDDoesNotExist(watcher, fooPathID2) {
-			return false
-		}
-
-		// Checking PID1 is still associated to the 2 paths, and PID2 is associated only with the first path
-		return checkPIDAssociatedWithPathID(watcher, fooPathID1, uint32(command1.Process.Pid)) &&
-			checkPIDAssociatedWithPathID(watcher, fooPathID2, uint32(command1.Process.Pid)) &&
-			checkPIDAssociatedWithPathID(watcher, fooPathID1, uint32(command2.Process.Pid)) &&
-			checkPIDNotAssociatedWithPathID(watcher, fooPathID2, uint32(command2.Process.Pid))
+		// Check that no more callbacks were executed, but we're tracking two PIDs now
+		return registerRecorder.CallsForPathID(fooPathID1) == 1 &&
+			registerRecorder.CallsForPathID(fooPathID2) == 1 &&
+			hasPID(watcher, command1) &&
+			hasPID(watcher, command2)
 	}, time.Second*10, time.Second, "")
 
 	require.NoError(t, command1.Process.Kill())
 	require.Eventuallyf(t, func() bool {
-		// Checking path1 still exists, and path2 not.
-		if checkPathIDDoesNotExist(watcher, fooPathID1) || checkPathIDExists(watcher, fooPathID2) {
-			return false
-		}
-
-		// Checking PID1 is not associated to the path 2, and PID2 is associated only with the path2
-		return checkPIDNotAssociatedWithPathID(watcher, fooPathID1, uint32(command1.Process.Pid)) &&
-			checkPIDAssociatedWithPathID(watcher, fooPathID1, uint32(command2.Process.Pid))
+		// Checking that the unregisteredCB was executed only for pathID2
+		return unregisterRecorder.CallsForPathID(fooPathID1) == 0 &&
+			unregisterRecorder.CallsForPathID(fooPathID2) == 1
 	}, time.Second*10, time.Second, "")
 
 	require.NoError(t, command2.Process.Kill())
 	require.Eventuallyf(t, func() bool {
-		// Checking path1 still exists, and path2 not.
-		return checkPathIDDoesNotExist(watcher, fooPathID1) && checkPathIDDoesNotExist(watcher, fooPathID2)
+		// Checking that the unregisteredCB was executed now for pathID1
+		return unregisterRecorder.CallsForPathID(fooPathID1) == 1
 	}, time.Second*10, time.Second, "")
-
-	checkWatcherStateIsClean(t, watcher)
```

Comment:
why did you remove that?
The test is intended to detect leaks, but now we cannot detect them if they exist

Reply:
Re-added check by ensuring that are no PIDs registered